charm_versions.md

```diff
@@ -0,0 +1,138 @@
+# Versions
+
+## Juju Version
+
+The Juju version uses [semantic versioning](https://semver.org/), so is a string in the format
+major.minor.patch, where each section is a positive integer (which may be higher than 9).
+
+### Agent
+
+You can see the version of Juju used in a model in `juju status`, in the top section under
+"Version" in the tabular view, or `model.version` in JSON or YAML output. You can also use the
+`juju model-config -m controller agent-version` command.
+
+To upgrade the patch version of Juju (e.g. 3.3.0 to 3.3.1), you run
+[`juju upgrade-controller`](https://juju.is/docs/juju/juju-upgrade-controller), specifying the
+patch version. To upgrade minor or major versions, you need to create a new controller and
+[migrate](https://juju.is/docs/juju/juju-migrate) the models to it.
+
+### CLI
+
+You can see the version of the Juju CLI tool using `juju --version`. When bootstrapping a new
+controller, this is the version of the agent that will be used, unless the `--agent-version` option
+is provided.
+
+To upgrade your local Juju version, use the same tool you used to install it. For example, if you
+installed Juju with snap, you can run `snap refresh juju --channel=3.3/stable` to upgrade to
+Juju 3.3.
+
+## Charms
+
+To upgrade the version or revision of an installed charm,
+[use `juju refresh`](https://juju.is/docs/juju/manage-charms#heading--update-a-charm) for both
+local and Charmhub charms.
+
+The [jhack](https://github.com/canonical/jhack) tool has a `charm vinfo` command that will show
+the revision, channel, and workload version of a charm, as well as the versions of each charm
+lib (and if there is a more recent version available).
+
```

Comment:
I am not sure if this is featured in the docs yet. But I agree it should be, though not sure where. jhack is a charm SDK tool, but here it provides an alternative to juju, but the alternative is designed to provide the level of detail that a charm author might need. Overall, still charm SDK material, but we don't have any clear place for it there yet -- let's think about it.